### ISSUE TITLE
VRAM address should only be set on successful register write pair

### ISSUE THREAD

**Issue by nrivard** [visrealm/vrEmuTms9918#3]
Current implementation can inadvertently erase the current vram address if you:
1. write a legitimate vram address to the VDP so it's pointing somewhere expected
2. write a value to register port (mode=1)
3. read status register (mode=1)
4. read data port (mode=0)

VDP has now overwritten lower byte of vram address. According to this [hardware deep-dive](http://bifi.msxnet.org/msxnet/tech/tms9918a.txt), changes are only committed on successful pairs.

**Comment by nrivard** [comment]
Locally, I have solved for this by adding a new field in the VDP struct:
```c
    /// if kVDPUnsetAddr (0xFFFF), register is in "reset" mode.
    /// if any other value, it is waiting for a value pair to complete the register write process
    uint16_t registerValue;
```
and my register write function:
```c
void VDPWriteToRegister(VideoDisplayProcessorRef vdp, uint8_t value) {
    if (!vdp) {
        return;
    }

    if (vdp->registerValue == kVDPUnsetAddr) {
        vdp->registerValue = value;
    } else {
        if (value & kVDPRegisterWriteMask) {
            // register write
            vdp->registers[value & (kVDPRegisterCount - 1)] = vdp->registerValue & 0xFF;
        } else {
            // vram address
            vdp->vramAddr = ((value & (kVDPVramWriteMask - 1)) << 8) | vdp->registerValue;
        }

        vdp->registerValue = kVDPUnsetAddr;
    }
}
```
Technically there's one more thing to do here, but I will write another issue for that 😄

**Comment by visrealm** [comment]
Closed as fixed with commit 56b100c
